### DISCUSSION TITLE
Pedantry related to TR's Raw Ore, or "Why do dusts craft into rocks?"

### DISCUSSION THREAD

**Discussion by Bluberry-Kat** [TechReborn/TechReborn#2543] [General]
(I couldn't decide if this should be a bug or feature report. "Feature" felt too presumptuous, but "Bug" implies must-fix. Alas.)  
Caves and Cliffs part 1 introduced Raw Ores, meaning "ore doubling" is now a sanctioned vanilla mechanic.  
Tech Reborn (this mod), seeking parity, has started registering its own Raw Ores, in place of what was previously dusts.  

However, some cracks are showing. Notably, small dusts fit in significantly less than they used to.
![4 Small Lead Dust crafting into a Raw Lead.](https://user-images.githubusercontent.com/20749209/124341705-48e7fe80-db8c-11eb-8edb-c70273138e79.png)

I wouldn't have made this issue if there weren't a relatively simple solution.  
Like how TR has changed dusts to raw ore, small dusts could be changed to raw pebbles.  
This would entail textures for the pebbles, updated localization, registry related refactors, and potentially even data-fixers, depending on how in-depth the translation could be.  
A simple job, but a significant amount of work. Might be worth waiting until 1.18?

As a far more pedantic sidenote, even in an Industrial Grinder, I don't think I agree with ingots being ground into raw ore.  
While Tech Reborn has turned dusts into raw ore, whose to say other mods will do the same? Maybe some mod will let you grind raw silver into dust, smelt it, and then with an Industrial Grinder they could turn it into raw ore again, repeating the process and getting infinite silver. These recipes do cost mercury now, but is it enough?  
In-mod, is there a merit to using raw ore this way? Particularly, in recipes that require it, like glass fibre cable, is there a problem having those just take ingots instead?
![Industrial Grinder recipe, using Mercury to grind a Silver Ingot into Raw Silver.](https://user-images.githubusercontent.com/20749209/124341854-72555a00-db8d-11eb-986e-1db65564e0d0.png)

I'm an experienced data-monkey and can quickly tweak large quantities of mojangson.  
If you'd like me to replace Raw Ore in non-ore-processing recipes with Ingots, I can get to work on a pull request. Let me know if there's interest.

**Comment by drcrazy** [comment]
I'm totally agree on small metal dusts falling out of line. Maybe we should use nuggets instead. But this will be even harder to balance compare to ore -> raw ore -> ingot chain. 

As for uncrafting recipe - I would like to keep it for completeness

Usage of ingots instead of raw ores - good point. It could be even plates, not just ingots.

**Comment by Bluberry-Kat** [comment]
> Maybe we should use nuggets instead. But this will be even harder to balance compare to ore -> raw ore -> ingot chain.  

Another option would be to _keep_ all of the dusts, and instead grinding a Raw Ore will produce maybe 5 small dust (or 1&¼ ingots).  
This would also resolve my nitpick regarding the uncrafting recipe — it could be moved back to producing a dust instead of a raw ore.

**Comment by Ayutac** [comment]
personally I am a fan of avoiding unnecessary items. Another solution would be to change the small dust piles of vanilla variants to nuggets, but keep the dust for TR ores. Not a solution I am a fan of, I would rather see the TR ores approaching the new vanilla ones.

**Comment by drcrazy** [comment]
Yeah, ore - raw meta - ingot - nugget is a thing for vanilla. I see no reason why we should have some metals with that line and other metals with ore - dust - ingot - small dust.

**Comment by Bluberry-Kat** [comment]
There's a small thematic incongruity if you move wholesale to raw ore — particularly machines like the Grinder and the Centrifuge "make more sense" when they produce dust.  
I think the compromise solution is turning raw ore into dust. It doesn't even need to be doubling, could just be 1:1.  
This lets your machines still make sense, lets you keep raw ore, and also gives you a handy ingot-equivalent for recipes that you can be confident is always _exactly one ingot._  

The downside, of course, is item bloat. If we're trying to trim down on that though, I think there's much better places to start.  

For the record, I support my proposed solution and also "move wholesale to raw ore + nuggets." No strong preference for either.

**Comment by Ayutac** [comment]
Hm. I have another idea. As much as I disagree with the folk crying out against 4 Small Piles -> Raw.

Is grinding ingots into raw ores needed for TR? If not, we can prevent the issue by deleting that and changing it so 4 Small Piles cannot be crafted into raw, but only directly burned into 1 ingot.

**Comment by pipich3** [comment]
is there a possibility to have a random items count in recipe output?
this may allow outputting a whole Plie / Raw metal / Ingot with some chance < 1 instead of Small Piles
e.g. i.Grinding of Sheldonite Ore can output a Raw Iridium / Iridium Ingot with ~1/4 chance instead of 2 Iridium Nuggets

P.S.: this is just an idea how to avoid unnecessary items such as Small Piles / Nuggets
